qa/tasks/s3tests.py

```diff
@@ -461,27 +543,81 @@ def task(ctx, config):
 
     log.debug('s3tests config is %s', config)
 
+    global sts_variable
     s3tests_conf = {}
-    for client in clients:
-        endpoint = ctx.rgw.role_endpoints.get(client)
-        assert endpoint, 's3tests: no rgw endpoint for {}'.format(client)
-
-        s3tests_conf[client] = ConfigObj(
-            indent_type='',
-            infile={
-                'DEFAULT':
-                    {
-                    'port'      : endpoint.port,
-                    'is_secure' : endpoint.cert is not None,
-                    'api_name'  : 'default',
-                    },
-                'fixtures' : {},
-                's3 main'  : {},
-                's3 alt'   : {},
-		's3 tenant': {},
-                'iam'      : {},
-                }
-            )
+
+    for client, client_config in config.items():
+        #This will be the structure of config file when you want to run webidentity_test (sts-test)
+        if 'sts_tests' in client_config and "TOKEN" in os.environ:
```

Comment:
This conditional is repetitive an unnecessary. You should assign the value of sts_variable before this loop so you don't need to define an empty `s3tests_conf` over and over again.